src/pybind/mgr/cephadm/upgrade.py

```diff
@@ -250,7 +251,20 @@ def continue_upgrade(self) -> bool:
         :return:
         """
         if self.upgrade_state and not self.upgrade_state.paused:
-            self._do_upgrade()
+            try:
+                self._do_upgrade()
+            except (OrchestratorError, ImportError, NotImplementedError) as e:
+                # these exception types will be handled by orchestrator
+                raise e
+            except Exception as e:
+                # other exception types can cause cephadm module to fail
+                self._fail_upgrade('UPGRADE_EXCEPTION', {
+                    'severity': 'error',
+                    'summary': f'Upgrade: failed due to exception: {str(e)}',
+                    'count': 1,
+                    'detail': ['Unexpected exception occurred during upgrade process'],
+                })
+                raise e
```

Comment:
will also break serve and cause the module to fail ... maybe return False here?

```suggestion
                return False
```

Reply:
So far I was just going to keep the functionality roughly the same while just providing more information. Do we want the module to keep going when we have exceptions like this that we don't know how to handle?

Reply:
I think we should, because an unexpected exception during upgrade could also break all the other orch commands ...